### PR TITLE
SNOW-1887537 Get file without SFC_DIGEST

### PR DESCRIPTION
**Pull request by sfc-gh-knozderko**
### Description
SNOW-1887537 Get file without SFC_DIGEST

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

**Comment by sfc-gh-pcyrek** [comment]
check the return value to ensure the correct number of bytes has been read and get rid off warnings, e.g.:
 
      int bytesRead = stream.Read(buf, 0, col1ToBytes.Length);
      if (bytesRead != col1ToBytes.Length)
      {
          throw new InvalidOperationException("Fewer bytes read than expected.");
      }

File: Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs [hunk 1]
```diff
@@ -190,11 +190,15 @@ private bool TryGetMetadataValueCaseInsensitive(BlobProperties properties, strin
             return keysCaseInsensitive.Any() ? properties.Metadata.TryGetValue(keysCaseInsensitive.First(), out metadataValue) : false;
         }
 
-        private string GetMetadataValueCaseInsensitive(BlobProperties properties, string metadataKey)
+        private string GetMetadataValueCaseInsensitive(BlobProperties properties, string metadataKey, bool failIfNotFound = true)
```
Comment (sfc-gh-pcyrek) [inline Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs:193]: Fail if not found - double negation,  

I will rather  go with 'allow empty' and adjust logic accordingly. Easier to follow.

File: Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs [hunk 2]
```diff
@@ -190,11 +190,15 @@ private bool TryGetMetadataValueCaseInsensitive(BlobProperties properties, strin
...
         {
             if (TryGetMetadataValueCaseInsensitive(properties, metadataKey, out var metadataValue))
                 return metadataValue;
-            throw new KeyNotFoundException($"The given key '{metadataKey}' was not present in the dictionary.");
+            if (failIfNotFound)
+            {
+                throw new KeyNotFoundException($"The given key '{metadataKey}' was not present in the dictionary.");
```
Comment (sfc-gh-pcyrek) [inline Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs:199]: Can we cover scenario with test where flag is active and response is empty?

File: Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs (hunk 1 above)
Comment (sfc-gh-knozderko) [reply Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs:193]: empty string is something different than null.
We want "allowEmpty" suggest something different that non existence of the peroperty.
I would preserve the current name.

File: Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs (hunk 2 above)
Comment (sfc-gh-knozderko) [reply Snowflake.Data/Core/FileTransfer/StorageClient/SFSnowflakeAzureClient.cs:199]: done.

**Review by sfc-gh-pmotacki — APPROVED** [review]

File: Snowflake.Data.Tests/UnitTests/SFAzureClientTest.cs [hunk 3]
```diff
@@ -410,5 +410,64 @@ public void TestEncryptionMetadataReadingIsCaseInsensitive()
...
+            // assert
+            Assert.AreEqual(ResultStatus.UPLOADED.ToString(), _fileMetadata.resultStatus);
+            Assert.IsNull(fileHeader.digest);
+            Assert.AreEqual("initVector", fileHeader.encryptionMetadata.iv);
+            Assert.AreEqual("key", fileHeader.encryptionMetadata.key);
+            Assert.AreEqual("description", fileHeader.encryptionMetadata.matDesc);
+        }
+
+        [Test]
+        public void TestEncryptionMetadataReadingFailsWhenNoMandatoryProperty()
```
Comment (sfc-gh-pfus) [inline Snowflake.Data.Tests/UnitTests/SFAzureClientTest.cs:447]: A verb missing at the end? `exists/isPresent`?

**Review by sfc-gh-pfus — APPROVED** [review]

**Review by sfc-gh-pcyrek — APPROVED** [review]

File: Snowflake.Data.Tests/UnitTests/SFAzureClientTest.cs (hunk 3 above)
Comment (sfc-gh-knozderko) [reply Snowflake.Data.Tests/UnitTests/SFAzureClientTest.cs]: done
